artifacts/deploy/karmada-metrics-adapter-apiservice.yaml

```diff
@@ -15,6 +15,34 @@ spec:
   version: v1beta1
   versionPriority: 10
 ---
+apiVersion: apiregistration.k8s.io/v1
+kind: APIService
+metadata:
+  name: v1beta2.custom.metrics.k8s.io
+spec:
+  service:
+    name: karmada-metrics-adapter
+    namespace:  karmada-system
+  group: custom.metrics.k8s.io
+  version: v1beta2
+  insecureSkipTLSVerify: true
```

Comment:
Why skip TLS verification here?

Reply:
I just found the previous `metrics.k8s.io` also skipped TLS verification. @jwcesign Why?

Reply:
Yeah, I know, that's exactly why I'm asking @jwcesign :)

Reply:
Hi, I mentioned was `karmada-clutser-aggregated-apiserver` instead of `metrics.k8s.io`.

Reply:
Then cc @XiShanYongYe-Chang

Reply:
The information I found:
1. If delete this field, there will be error in `karmad-apiserver`:
 ```text
E0615 06:35:09.057036       1 controller.go:116] loading OpenAPI spec for "v1alpha1.cluster.karmada.io" failed with: failed to retrieve openAPI spec, http error: ResponseCode: 503, Body: error trying to reach service: x509: certificate signed by unknown authority
```

2. If I use curl pod to access the service url `https://karmada-aggregated-apiserver.karmada-system.svc.cluster.local`
```text
root@karmada [02:36:46 PM] [~/workspace/git]
-> # k exec -nkarmada-system curl-6c86467dcd-577h2 -- curl https://karmada-aggregated-apiserver.karmada-system.svc.cluster.local
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
curl: (60) SSL certificate problem: unable to get local issuer certificate
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
command terminated with exit code 60
```

3.If I use curl pod to access the service url `https://karmada-aggregated-apiserver.karmada-system.svc.cluster.local` and specify the ca.crt:
```text
root@karmada [02:38:01 PM] [~/workspace/git]
-> # k exec -nkarmada-system curl-6c86467dcd-577h2 -- curl --cacert /etc/karmada/pki/ca.crt https://karmada-aggregated-apiserver.karmada-system.svc.cluster.local
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {},
  "status": "Failure",
  "message": "forbidden: User \"system:anonymous\" cannot get path \"/\"",
  "reason": "Forbidden",
  "details": {},
  "code": 403
100   217  100   217    0     0  18523      0 --:--:-- --:--:-- --:--:-- 19727
```

So I think the problem is in `karmada-apiserver`, it validate aggregated-apiserver 's certificate with wrong ca.

Reply:
![image](https://github.com/karmada-io/karmada/assets/30589999/0485994b-efc2-4bac-a8ed-c4c380b1a85a)

Reply:
you should specify caBundle, then the kube-apiserver can known using which CA to verify the certificate of extension-apiserver.

`insecureSkipTLSVerify` is not advised.

If neither `insecureSkipTLSVerify` nor `caBundle` was set, system trust roots on the apiserver are used. This can cause certificate authentication to fail, which is the another CA certificate that you speculate about.

you can find more message in: https://kubernetes.io/zh-cn/docs/reference/kubernetes-api/cluster-resources/api-service-v1/

Reply:
So can we mount the ca in secret as the ca of the `karmada-apiserver` container?

Reply:
> So can we mount the ca in secret as the ca of the `karmada-apiserver` container?

1. ca is actually public key, not sensitive information, you can write our own CA in the `caBundle field`.
2. the type of `caBundle field` is `[]byte`, so you should write Base64 encoded plaintext CA data, you may not be able to use the method of mounting secret.

Reply:
For now, the `ca.crt` is located in the `karmada-secret`. We cannot fill the `caBundle` field with secret data. Therefore, if users wish to enable secure TLS, they must manually fill this field. CC: @XiShanYongYe-Chang @RainbowMango

Reply:
Please open an issue to track this, so that we can move this PR forward.

Reply:
done: https://github.com/karmada-io/karmada/issues/3672